### PR TITLE
Vcfg add instance name

### PR DESCRIPTION
**Pull request by sgilbride**
Added prompt for instance name to vcfg in all cases.
Updated tests to match.

### CODE REVIEW

File: volttron/platform/instance_setup.py [hunk 1]
```diff
@@ -484,6 +484,23 @@ def do_vip():
             print('\nERROR: That address has already been bound to.')
     config_opts['vip-address'] = '{}:{}'.format(vip_address, vip_port)
 
+
+def do_instance_name():
+    global config_opts
+    # Default instance name to the vip address.
```
Comment (craig8) [inline volttron/platform/instance_setup.py:490]: this comment doesn't make sense.

File: volttron/platform/instance_setup.py [hunk 2]
```diff
@@ -484,6 +484,23 @@ def do_vip():
...
+    instance_name = config_opts.get('instance-name',
+                                    'volttron1')
+    instance_name = instance_name.strip('"')
+
+    valid_name = False
+    while not valid_name:
+        prompt = 'What is the name of this instance?'
+        new_instance_name = prompt_response(prompt, default=instance_name)
+        if new_instance_name:
```
Comment (craig8) [inline volttron/platform/instance_setup.py:499]: I believe there is more to validating an instance name than just a non-empty string.  We need to discuss this.

File: volttrontesting/platform/test_instance_setup.py [hunk 3]
```diff
@@ -52,9 +52,10 @@
 def create_vcfg_vhome():
     debug_flag = os.environ.get('DEBUG', False)
     vhome = create_volttron_home()
-    yield vhome
+    vhome_dir = os.path.split(vhome)[0]
```
Comment (craig8) [inline volttrontesting/platform/test_instance_setup.py:55]: ummm....gonna have to tell me about this one?

**Review by craig8 — COMMENTED** [review]
Lets discuss in a breakout room today

File: volttron/platform/instance_setup.py (hunk 2 above)
Comment (sgilbride) [reply volttron/platform/instance_setup.py:504]: JIRA issue has been submitted to discuss this further. No change at this time.

File: volttron/platform/instance_setup.py (hunk 1 above)
Comment (sgilbride) [reply volttron/platform/instance_setup.py]: Addressed.

File: volttrontesting/platform/test_instance_setup.py (hunk 3 above)
Comment (sgilbride) [reply volttrontesting/platform/test_instance_setup.py]: Fixed.

**Review by shwethanidd — APPROVED** [review]

**Pull request merged**